storage/include/blockchain/merkle_tree_serialization.h

```diff
@@ -345,6 +350,22 @@ inline block::detail::Node deserialize<block::detail::Node>(const concordUtils::
   return node;
 }
 
+// Deserializes the state root version from a serialized block::detail::Node .
+inline sparse_merkle::Version deserializeStateRootVersion(const concordUtils::Sliver &buf) {
+  Assert(buf.length() >= block::detail::Node::MIN_SIZE);
+  constexpr auto offset = sizeof(block::detail::Node::BlockIdType) + block::detail::Node::PARENT_DIGEST_SIZE +
```

Comment:
That's a good point and I have also been thinking about it. I agree there is room for improvement here.

An option to consider is to introduce `serialize()` and `deserialize()` methods to types. That will solve the dependency between files. However, it might be a big burden on types that have a different purpose altogether. That is why I only introduced `SIZE_IN_BYTES` members in `sparse_merkle` types and not code or offsets. On the other hand, why would anybody care about size if not for serialization...

Another option might be to remove public data members and enforce usage of constructors so that users can't create an object in an incomplete state. Then, as you mention, we already have tests for serialization and that would help us find issues when there is a breaking change.

The problem is more relevant for types that are not part of the `detail` namespace, e.g. types in the `sparse_merkle` namespace as it would be perfectly valid to change these types without thinking about serialization there (and I think people should be encouraged not to think about it there).

Let me leave as is in this PR and please let's discuss in order to find a solution that will solve it for all cases in a uniform manner.

Thanks!

Reply:
Ok. Cool. I'm also not sure I want to introduce serialize()/deserialize() methods. We can discuss at a later point.